theano/scalar/basic_scipy.py

```diff
@@ -542,3 +544,30 @@ def grad(self, inp, grads):
         return [gz * i1(x)]
 
 i0 = I0(upgrade_to_float, name='i0')
+
+
+class Polygamma(UnaryScalarOp):
+    """
+    (m-1)'th derivative of gammaln
+    """
+    __props__ = ('m', )
+
+    def __init__(self, m, *args, **kwargs):
+        super(Polygamma, self).__init__(*args, **kwargs)
+        self.m = m
+
+    def impl(self, x):
+        if imported_scipy_special:
+            return scipy.special.polygamma(self.m, x)
+        else:
+            # raises an error
+            super(Polygamma, self).impl(x)
+
+    def grad(self, inp, grads):
+        x, = inp
+        gz, = grads
+        # derivative is again polygamma
+        poly = Polygamma(self.m+1, upgrade_to_float, name='_%dgamma' % (self.m+1))
```

Comment:
I'm not sure. If we want to compute the polygamma of x and the derivative. Can you reuse some of the forward computation? Is there a numpy/scipy function that does that?

The way you do it looks fine. What would you think could be made better?